cmd/monitor/monitor.go

```diff
@@ -307,9 +317,7 @@ func (ms *monitorStatus) getBlockRange(ctx context.Context, from, to *big.Int, r
 		}
 		pb := rpctypes.NewPolyBlock(b.Result.(*rpctypes.RawBlockResponse))
 
-		ms.BlocksLock.Lock()
-		ms.Blocks[pb.Number().String()] = pb
-		ms.BlocksLock.Unlock()
+		ms.BlockCache.Add(pb.Number().String(), pb)
```

Comment:
- The Add method checks if the cache is at its maximum capacity.
- If the cache is full, it evicts the least recently used item before adding the new item.
- If the cache is not full, it simply adds the new item to the cache.

Reply:
Maybe you could add this in the code as a comment also?